plutus-metatheory/src/Algorithmic/ReductionEC.lagda.md

```diff
@@ -666,11 +666,15 @@ bappTermLem mkNilData _ (start _) (base refl) | refl ,, refl = _ ,, _ ,, refl
 bappTermLem mkNilPairData {az = az} {as} M p q
   with <>>-cancel-both az ([] :< Term) as p
 bappTermLem mkNilPairData _ (start _) (base refl) | refl ,, refl = _ ,, _ ,, refl
-bappTermLem mkCons _ (start _) (base refl) = _ ,, _ ,, refl
-bappTermLem mkCons {as = as} _ (bubble {as = az} p) q
-  with <>>-cancel-both' az _ (([] :< Term) :< Term) as p refl
-bappTermLem mkCons _ (bubble (start _)) (step _ (base refl) _)
-  | refl ,, refl ,, refl = _ ,, _ ,, refl
+bappTermLem mkCons _ (bubble (start _)) (step⋆ _ (base refl) refl) =
+  _ ,, _ ,, refl
+bappTermLem mkCons _ (bubble (bubble {as = az} p)) q
+  with <>>-cancel-both' az _ ((([] :< Type) :< Term) :< Term) _ p refl
+bappTermLem mkCons
+            _
+            (bubble (bubble (start _)))
+            (step _ (step⋆ _ (base refl) refl) x)
+            | refl ,, refl ,, refl = _ ,, _ ,, refl
```

Comment:
Do you have any plans on deduplicating this too? We seem to have the same in the `CEKV` module.

Reply:
I think there's a ticket for that.

Reply:
Yes, definitely.  No only deduplicating them, but making them generic on the builtin.